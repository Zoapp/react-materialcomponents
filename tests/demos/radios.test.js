import React from "react";
import renderer from "react-test-renderer";
import Radios from "demos/components/radios";

describe("demos/components/radios", () => {
  it("renders correctly", () => {
    const component = renderer.create(<Radios />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
