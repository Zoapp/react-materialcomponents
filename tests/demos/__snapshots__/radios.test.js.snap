// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demos/components/radios renders correctly 1`] = `
<section>
  <h1>
    Radio examples 
  </h1>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="mdc-form-field"
    >
      <div
        className="mdc-radio"
      >
        <input
          className="mdc-radio__native-control"
          defaultChecked="checked"
          id="1"
          name="r1"
          type="radio"
        />
        <div
          className="mdc-radio__background"
        >
          <div
            className="mdc-radio__outer-circle"
          />
          <div
            className="mdc-radio__inner-circle"
          />
        </div>
      </div>
      <label
        htmlFor="1"
      >
        radio 1
      </label>
    </div>
    <div
      className="mdc-form-field"
    >
      <div
        className="mdc-radio"
      >
        <input
          className="mdc-radio__native-control"
          id="2"
          name="r1"
          type="radio"
        />
        <div
          className="mdc-radio__background"
        >
          <div
            className="mdc-radio__outer-circle"
          />
          <div
            className="mdc-radio__inner-circle"
          />
        </div>
      </div>
      <label
        htmlFor="2"
      >
        radio 2
      </label>
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="mdc-form-field"
    >
      <div
        className="mdc-radio"
      >
        <input
          className="mdc-radio__native-control"
          defaultChecked="checked"
          id="3"
          name="r2"
          type="radio"
        />
        <div
          className="mdc-radio__background"
        >
          <div
            className="mdc-radio__outer-circle"
          />
          <div
            className="mdc-radio__inner-circle"
          />
        </div>
      </div>
      <label
        htmlFor="3"
      >
        radio 1
      </label>
    </div>
    <div
      className="mdc-form-field"
    >
      <div
        className="mdc-radio"
      >
        <input
          className="mdc-radio__native-control"
          id="4"
          name="r2"
          type="radio"
        />
        <div
          className="mdc-radio__background"
        >
          <div
            className="mdc-radio__outer-circle"
          />
          <div
            className="mdc-radio__inner-circle"
          />
        </div>
      </div>
      <label
        htmlFor="4"
      >
        radio 2
      </label>
    </div>
  </div>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      className="mdc-form-field"
    >
      <div
        className="mdc-radio"
      >
        <input
          className="mdc-radio__native-control"
          defaultChecked="checked"
          disabled="disabled"
          id="5"
          name="r3"
          type="radio"
        />
        <div
          className="mdc-radio__background"
        >
          <div
            className="mdc-radio__outer-circle"
          />
          <div
            className="mdc-radio__inner-circle"
          />
        </div>
      </div>
      <label
        htmlFor="5"
      >
        radio 1
      </label>
    </div>
    <div
      className="mdc-form-field"
    >
      <div
        className="mdc-radio"
      >
        <input
          className="mdc-radio__native-control"
          disabled="disabled"
          id="6"
          name="r3"
          type="radio"
        />
        <div
          className="mdc-radio__background"
        >
          <div
            className="mdc-radio__outer-circle"
          />
          <div
            className="mdc-radio__inner-circle"
          />
        </div>
      </div>
      <label
        htmlFor="6"
      >
        radio 2
      </label>
    </div>
  </div>
  <h2>
    Radio Button Properties
  </h2>
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
        style={
          Object {
            "fontSize": "20px",
            "fontWeight": "bold",
          }
        }
      >
        RadioButtonProperties
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
      >
        Name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Default
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Type
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        checkedIcon
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        element
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        The icon element to show when the radio button is checked.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        disabled
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        If true,  the radio button is disabled.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        iconStyle
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        object
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Override the inline-styles of the icon element.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        inputStyle
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        object
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Override the inline-styles of the input element.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        labelStyle
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        object
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Override the inline-styles of the label element.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        style
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        object
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Override the inline-styles of the root element.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        uncheckedIcon
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        element
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        The icon element to show when the radio button is unchecked.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        value
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        any
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        The value of the radio button.
      </div>
    </div>
  </div>
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
        style={
          Object {
            "fontSize": "20px",
            "fontWeight": "bold",
          }
        }
      >
        RadioButtonGroupProperties
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
      >
        Name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Default
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Type
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        children
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        node
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Should be used to pass RadioButton components.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        className
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        The CSS class name of the root element.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        defaultSelected
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        any
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        The value property of the radio button that will be selected by default. This takes precedence over the checked property of the RadioButton elements.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        labelPosition
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        enum:'left', 'right'
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Where the label will be placed for all child radio buttons. This takes precedence over the labelPosition property of the RadioButton elements.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        The name that will be applied to all child radio buttons.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        onChange
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        function
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Callback function that is fired when a radio button has been checked.Signature: function(event: object,  value: undefined) =&gt; void event: change event targeting the selected radio button. value: The value of the selected radio button.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        style
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        object
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Override the inline-styles of the root element.
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        valueSelected
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        any
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        The value of the currently selected radio button.
      </div>
    </div>
  </div>
</section>
`;
